resources/scripts/cycle-diff.sh

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env bash
+
+# Performance Comparison Script
+# ------------------------------
+#
+# This script performs a performance comparison between the current branch and a base branch
+# (default: nightly) for a specified test. It can generate comparison results and check for
+# performance regressions.
+#
+# Usage:
+#   ./cycle-diff.sh [generate|check]
+#
+# Modes:
+#   generate: Runs tests on both current and base branches, generates comparison results.
+#   check: Checks for performance regression based on pre-generated results.
+#
+# Environment Variables:
+#   BASE_BRANCH: The branch to compare against (default: nightly)
+#   TEST_NAME: The test file to run (default: bitcoin_e2e::tests::prover_test::basic_prover_test)
+#   TARGET_PCT: The threshold percentage for regression detection (default: 3)
+#
+# Requirements:
+#   - Git
+#   - Rust and Cargo
+#
+# WARNING: This script will checkout to BASE_BRANCH. Ensure you are working with a clean
+#          Git repository (all changes committed) before running this script.
+#
+# Output:
+#   - In 'generate' mode: Creates a file '$COMPARISON_FILE' with performance metrics.
+#   - In 'check' mode: Outputs whether a performance regression was detected.
+#
+# Exit Codes:
+#   0: Success or no regression detected
+#   1: Error or regression detected
+
+set -euo pipefail
+
+BASE_BRANCH=${BASE_BRANCH:-"nightly"}
+TEST_NAME=${TEST_NAME:-"bitcoin_e2e::tests::prover_test::basic_prover_test"}
+TARGET_PCT=${TARGET_PCT:-3}
+COMPARISON_FILE=${COMPARISON_FILE:-"comparison_results.log"}
+
+# Used to silence custom risc0 build.rs output
+export RISC0_GUEST_LOGFILE=$(mktemp)
+
+run_test_and_extract() {
+    local command="cargo test $TEST_NAME -- --nocapture"
+    local output_file="output.log"
+
+    # Aggressively silence build output
+    script -q /dev/null make build > /dev/null 2>&1
```

Comment:
`make build` will build with profile `dev`, then `cargo test` will rebuild it with profile `test`.

Reply:
Why not introduce `make release` which builds with `--release` since we want to optimize as much as possible for the cycle counts?

Reply:
guest code is built with a separate build.rs script which builds it in `--release` anyway.